src/physics/material/MaterialView.i.hh

```diff
@@ -69,13 +69,19 @@ CELER_FUNCTION ElementView MaterialView::element_view(ElementComponentId id) con
 
 //---------------------------------------------------------------------------//
 /*!
- * Total number density of an element in this material [1/cm^3]
+ * Number density of an element in this material [1/cm^3]
+ * 
+ * Number density n_a_i of i-th material is
+ * n_a_i = N_A * \rho * f_i / A_i,
+ * where \rho is the material density and f_i is the elemental mass fraction
  */
 CELER_FUNCTION real_type
 MaterialView::get_element_density(ElementComponentId id) const
 {
     REQUIRE(id < this->material_def().elements.size());
-    return this->number_density() * this->elements()[id.get()].fraction;
+    return constants::na_avogadro * this->density()
+           * this->elements()[id.get()].fraction
+           / this->element_view(id).atomic_mass().value();
```

Comment:
As I type this we're having our late-night slack discussion about the confusion of `fraction` being mass vs number... this should be reverted and I'll review again after we sort out the input conversion stuff.